File: 199-binary-tree-right-side-view/199-binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        q = []
        res = []
        if root == None:
            return res
        q.append(root)
        while len(q) > 0:
            size = len(q)
            res.append(q[0].val)
            for i in range(size):
                curr = q.pop(0)
                if curr.right:
                    q.append(curr.right)
                if curr.left:
                    q.append(curr.left)
        return res
                
                
            

File: 199-binary-tree-right-side-view/199-binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> res = new ArrayList<>();
        
        if(root == null)
            return res;
        Queue<TreeNode> q = new LinkedList<>();
        q.add(root);
        while(!q.isEmpty()){
            int size = q.size();
            for(int i=0;i<size;i++){
                TreeNode p = q.poll();
                if(i==0)
                    res.add(p.val);
                if(p.right != null){
                    q.add(p.right);
                }
                if(p.left != null){
                    q.add(p.left);
                }
            }
        }
        return res;
    }
}